tools/toolchain/clang_lto_pgo.sh

```diff
@@ -0,0 +1,143 @@
+    ninja -C build
+
+    USE_NEW_COMPILER=(-DCMAKE_C_COMPILER="$STAGE0_BIN/clang" -DCMAKE_CXX_COMPILER="$STAGE0_BIN/clang++" -DLLVM_USE_LINKER="$STAGE0_BIN/ld.lld")
+    NEW_COMMON_OPTS=(-DLLVM_ENABLE_PROJECTS="clang;lld" -DLLVM_ENABLE_RUNTIMES="compiler-rt" -DLLVM_ENABLE_LTO=Thin -DCLANG_DEFAULT_PIE_ON_LINUX=OFF -DLLVM_BUILD_TOOLS=OFF -DCMAKE_C_FLAGS="-Xclang -mllvm -Xclang -vp-counters-per-site=16" -DCMAKE_CXX_FLAGS="-Xclang -mllvm -Xclang -vp-counters-per-site=16")
+
+    echo "stage-1: Build a PGO-optimized compiler using the boostrapped compiler"
+
+    # Build a PGO-optimized compiler using the boostrapped compiler.
+    rm -rf ../stage-1-${CLANG_ARCH}
+    git fetch --depth=1 origin tag llvmorg-${LLVM_SCYLLA_TAG}
+    git worktree add ../stage-1-${CLANG_ARCH} llvmorg-${LLVM_SCYLLA_TAG}
+    cd ../stage-1-${CLANG_ARCH}
+    cmake -B build -S llvm "${USE_NEW_COMPILER[@]}" "${COMMON_OPTS[@]}" "${NEW_COMMON_OPTS[@]}" ${LLVM_TARGET} -DLLVM_BUILD_INSTRUMENTED=IR
+    ninja -C build
+
+    echo "cmake stage-1 output:"
+
+    cd ../
+    # Download the scylla codebase for training.
+    # Scylla's dependencies are already installed.
+    rm -rf scylla-${CLANG_ARCH}
+    mkdir scylla-${CLANG_ARCH}
+    cd scylla-${CLANG_ARCH}
+    tar -xpf $DIR/scylla-archive.tar
+
+    # 1st ScyllaDB compilation: gather a clang profile for PGO
+    cd ../scylla-${CLANG_ARCH}
+    rm -rf build build.ninja
+    ./configure.py "${SCYLLA_OPTS[@]}"
+    LLVM_PROFILE_FILE=$DIR/stage-1-${CLANG_ARCH}/build/profiles/default_%p-%m.profraw ninja build
+
+    cd ../stage-1-${CLANG_ARCH}
+    $STAGE0_BIN/llvm-profdata merge $DIR/stage-1-${CLANG_ARCH}/build/profiles/default_*.profraw -output=ir.prof
+    rm -r build
+    cmake -B build -S llvm "${USE_NEW_COMPILER[@]}" "${COMMON_OPTS[@]}" "${NEW_COMMON_OPTS[@]}" "${LLVM_TARGET}" -DLLVM_BUILD_INSTRUMENTED=CSIR -DLLVM_PROFDATA_FILE=$(realpath ir.prof)
+    ninja -C build
+
+    # 2nd compilation: gathering a clang profile for CSPGO
+    cd ../scylla-${CLANG_ARCH}
+    rm -rf build build.ninja
+    ./configure.py "${SCYLLA_OPTS[@]}"
+    LLVM_PROFILE_FILE=$DIR/stage-1-${CLANG_ARCH}/build/profiles/csir-%p-%m.profraw ninja build
+
+    cd ../stage-1-${CLANG_ARCH}
+    $STAGE0_BIN/llvm-profdata merge build/csprofiles/default_*.profraw -output=csir.prof
+    $STAGE0_BIN/llvm-profdata merge ir.prof csir.prof -output=combined.prof
+    rm -r build
+    # -DLLVM_LIBDIR_SUFFIX=64 for Fedora compatibility
+    cmake -B build -S llvm "${USE_NEW_COMPILER[@]}" "${COMMON_OPTS[@]}" "${NEW_COMMON_OPTS[@]}" -DLLVM_PROFDATA_FILE=$(realpath combined.prof) -DCMAKE_EXE_LINKER_FLAGS="-Wl,--emit-relocs" -DCMAKE_INSTALL_PREFIX=/usr -DLLVM_LIBDIR_SUFFIX=64
+    ninja -C build
+
+    #XXX: skipping BOLT for arm64 for now, since it causes segfault
```

Comment:
```suggestion
    #TODO: skipping BOLT for arm64 for now, since it causes segfault
```

Reply:
Does it still segfault?

Reply:
At least on my test on clang-17.0.6, it segfault on arm64.